### ISSUE TITLE
no check for overpay on house buying

### ISSUE THREAD

**Issue by Warloxx01** [ACEmulator/ACE#1723]
### Bug Report or Feature Request (mark with an `x`)
```
- [ X] bug report -> please search issues before submitting
- [ ] feature request
```

### Repro steps.
when buying a house place the special item, Writ and 10 M note or any amount larger then the exact amount in the buy window. it will take all the items, including all 10 M Notes. 

### The log given by the failure.
No log Found

### Desired functionality.
Should be able to buy a house and drop any amount say 2 MMDs and have it return the unused/unneeded amount.

What would like to see implemented?
What is the use case?
-->

**Comment by gmriggs** [comment]
I think this is how it was in retail too? Maybe someone can confirm..

I agree this would be a nice (possibly optional) convenience feature though

**Comment by Warloxx01** [comment]
I could be wrong on the purchase of the housing. Im still looking into it. but Im sure that it gave back change (pyreal) if you over paid. I know for sure that paying Maintenance it diffidently gave back change. I paid the Maintenance fees for 3 cottages in Lillyglen and I would drop 2 stacks of pyreal on it and it asked me if I wanted to pay the fee for someone else's cottage and I would say yes and it would leave a stack of 20k in my pack and so on till I was at 10k in the one stack. 25k + 25k - 30k = 20k and so on.

**Comment by Warloxx01** [comment]
LOL just found this : https://asheron.fandom.com/wiki/Housing

Purchasing a House

Each house in Asheron's Call has a purchase price that is made up of Pyreals, Trophies and one or more Writs of Refuge. In addition, each house also requires upkeep on a monthly basis (with the exception of Residential Quarters, which require upkeep every 3 months). One thing to make note of: If you own a cottage, villa or mansion, and you abandon it, you may not purchase another facility for 30 days.

Note: A list of the required trophies can be found on each individual house page.
Writs of Refuge

A new form of currency was introduced to Asheron's Call for the purpose of purchasing and maintaining houses. The Writ of Refuge is an item granted by members of the Zaikhal Arcanum for performing certain tasks or retrieving and returning certain trophy items to them. These tasks are geared roughly toward the level of the purchaser of the home at each level.

Arcanum Agents can be found in most cities and towns. Stop in and talk to one and he or she will tell you what items can be traded for writs. It is also possible to obtain writs through certain quests.
The Covenant Crystal

Covenant Crystals appear wherever there is a home. It is used to purchase and pay maintenance on the home. They link the character to the home and display the name of the homeowner once the dwelling has been purchased.

The covenant crystal accepts either pyreals or exact trade notes. For example, if purchasing a Villa you can use 2,000,000 pyreals or 8 MMD Notes.

After purchase of a dwelling, the first period's rent/maintenance is due, which is also paid at the covenant crystal. The next due date can be found on the Housing tab of the Map panel.

**Comment by gmriggs** [comment]
ah ok that might make sense, i think i read somewhere that it did not refund trade note overages, but that makes sense if it did refund overages for other items

**Comment by dirtyelf** [comment]
yes housing would refund pyreal stacks but not trade notes

**Comment by LtRipley36706** [comment]
Fixed with #2513
